applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+ * @param context        Pointer to the private context for the module
+ *
+ * @return 0 if successful, error value
+ */
+int amod_open(struct amod_parameters *parameters, struct amod_configuration *configuration,
+	      char *name, struct amod_handle *handle, struct amod_context *context);
+
+/**
+ * @brief  Function to close an open module.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_close(struct amod_handle *handle);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules configuration to set (as
+ *                       passed to the modules #amod_query_resource())
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_set(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules current configuration
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_get(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief Function to connect two modules together.
+ *
+ * @param handle_from  The handle for the module for output
+ * @param handle_to    The handle of the module to for input, if the same as ::handle from
+ *                     the data will be put on the ::handle_from output message quere.
+ *
+ * @return 0 if successful, error value
+ */
+int amod_connect(struct amod_handle *handle_from, struct amod_handle *handle_to);
+
+/**
+ * @brief Function to disconnect a module.
+ *
+ * @param handle             The handle for the module
+ * @param handle_disconnect  The handle of the module to disconnect
+ *
+ * @return 0 if successful, error value
+ */
+int amod_disconnect(struct amod_handle *handle, struct amod_handle *handle_disconnect);
+
+/**
+ * @brief Start a module processing data.
+ *
+ * @param handle  The handle for the module to start
+ *
+ * @return 0 if successful, error value
+ */
+int amod_start(struct amod_handle *handle);
+
+/**
+ * @brief Stop a module processing data.
+ *
+ * @param handle  The handle for the module to be stopped
+ *
+ * @return 0 if successful, error value
+ */
+int amod_stop(struct amod_handle *handle);
+
+/**
+ * @brief Send a data buffer to a module, all data is consumed by the module.
+ *
+ * @param handle       The handle for the receiving module instance
+ * @param block       Pointer to the audio data block to send to the module
+ * @param response_cb  A pointer to a callback to run when the buffer is
+ *                     fully comsumed in a module
+ *
+ * @return 0 if successful, error value
+ */
+int amod_data_tx(struct amod_handle *handle, struct aobj_block *block,
+		 amod_response_cb response_cb);
```

Comment:
response_cb should be defined when configuring/defining the module, and not here?

Reply:
I don't agree, this is a standard interface that is called when the data block has been finished with. It may not be the same for every call, even to the same module.